Beta 1.6 Release

hoffmang9 opened the pull request:
## Beta 1.6

### Added

- Windows and MacOS now have one click installers that then send users to a GUI on both platforms to farm or use their wallets. Windows is built on GitHub Actions and MacOS is also built on Azure Pipelines so as to build on Mojave.
- You can see and control your farmer, harvester, and plotter from the GUI on Windows, MacOS, and Linux.
- Create plots and see the plotting log from a GUI on Windows, MacOS, and Linux.
- You can now create or import private keys with a 24 word mnemonic, both in the UI and 'chia keys' command line.
- You can delete and change active keys from the GUI and cli.
- We added a new keychain system that replaces keys.yaml, and migrates existing users from keys.yaml. It utilizes each OS's keychain for slightly more secure key storage.
- We added a `chia keys` command line program, to see, add, and remove private keys.
- We added RPC servers and RPC client implementations for Farmer and Harvester. The new UI uses these for additional information and functionality.
- We added total network storage space estimation to the node RPC at the `/get_network_space` endpoint instead of only being available in the cli. The RPC endpoint takes two block header hashes and estimates space between those header hashes.
- We now have a CHANGELOG.md that adheres closely to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) standard. We merged in the version history and updated some previous release notes to capture items important to the change log. We are modifying our release process to accumulate changes at the top of the change log and then copy those to the release notes at the time of the release.
- We added [lgtm](https://lgtm.com/) source analysis on pull request to the chia-blockchain, chiapos, chiavdf, chiabip158, and bls-library repositories to add some automated security analysis to our ci.

### Changed

- Due to an issue with aggsig and aggsig-me, the beta 1.6 blockchain is not compatible with earlier chains.
- We replaced the Electron/JavaScript interface with a React user interface which is cleaner and more responsive.
- We now have a multithreaded harvester to farm more plots concurrently. This is especially faster when there are multiple disks being harvested. The class is also made thread safe with mutex guards. This is achieved by releasing GIL in the python bindings when fetching qualities and proofs. We estimate that the former guidance of only 50 plots per physical drive should be updated to 250-350 plots per physical drive. We will continue to improve the plots per physical drive limit during the beta period.
- Syncing a node is now much faster and uses less memory.
- `chia netspace` has been refactored to use the `/get_network_space` RPC. The command
  syntax has changed slightly. By default it calculates the last 24 blocks from the
  current LCA. Optionally you can use the `-b` flag to start the calculation from a different block
  height. Use `-d` to specify the delta number of blocks back into history to estimate over from either LCA or your `-b` block height.
- The Full node RPC response formats have been changed. All API calls now return a dict with success, and an additional value, for example {"success": True, "block": block}.
- chiapos is now easier to compile with MSVC.
- create plots now takes in an optional sk_seed, it is no longer read in from keys.yaml. If not passed in, it is randomly generated. The -i argument can now only be used when you provide an sk_seed.
- Moved to PyYAML 5.3.1 which prevents arbitrary code execution during python/object/new constructor.
- Moved to Python cryptography 2.9.2 which deprecates OpenSSL 1.0.1 and now relies upon OpenSSL 1.1.1g.
- Moved to aiosqlite 0.13.0 which adds official support for Python 3.8 and fixes a possible hung thread if a connection failed.

### Fixed

- In beta 1.5 we introduced a bug in aggsig and aggsig-me that we have fixed in this release. That forced a hard fork of the chain so coins and balances are lost from beta 1.5. There is no impact on existing plots.
- Starting and stopping servers now works much more reliably.
- `chia-check-plots` uses the plot root and checks the plots in the same manner as harvester.
- Fixed and issue where [Relic](https://github.com/relic-toolkit/relic) and thus blspy would crash on processors older than Haswell as they don't support lzc.
- Some non-critical networking errors are no longer logged.
- Blocks with compact proofs of time are now able to be updated into the node database.
- The `install-timelord.sh` script now correctly determines which version of python it is running under and correctly builds vdf_client and correctly links to vdf_bench. It also handles upgrading CMake on Ubuntu's older than 20.04LTS do satisfy the new CMake 3.14+ requirement to build Timelord.
- The build status shield layout is fixed in README.md
- Raspberry Pi 3/4 with Ubuntu 20.04LTS 64 bit should compile again.

hoffmang9 committed:
Oops. Fix README.md

wjblanke committed:
updated bls for lzc

wjblanke committed:
updated blspy

hoffmang9 committed:
Add logo to README.md

wjblanke committed:
update chiapos to 24

hoffmang9 committed:
Add CHANGELOG.md and adopt keepachangelog.com practices

hoffmang9 committed:
Bring CHANGELOG current and into conformance

hoffmang9 committed:
Change cli to use "Latest Common Ancestor"

mariano54 committed:
Ms.sync full sync improvements (#218)

* Refactor blockchain.py into a few different files (difficulty_adjustment.py, block_header_validation.py)
* Faster sync pipelining of requests, and parallel verification (13 minutes to 9 minutes on 2 cores)
* Memory usage of node during sync is significantly reduced, only the latest blocks are in memory
* Networking fixes and logging levels changed
* SPECIFIC delivery which allows you to send messages to a specific node id
* Check plots acts like the normal harvester

richardkiss committed:
Improve start (#217)

* Checkpoint.

* Add start_daemon_client.

* First crack at `chia daemon`.

* Improve UI.

* More improvements.

* Fix stop problem with groups.

* Simplify stop.

* Having problems with argparse.

* Try hard to prevent multiple chia daemon instances from running.

* Add a semaphore.

* Use UNIX sockets on platforms that have them.

* Fix problem with UNIX sockets.

* Remove lockfile on Windows.

* API server is now an aiter.

* Don't print anything when launching chia_farmer.

* Validate services before starting.

* Add comments.

* Close lockfile. Deal with subprocess launching errors better.

* Move service_groups file.

* Remove pid_path_killed if it exists before renaming.

* Bind to localhost.

* Fix flake8.

* Fix some PEP8.

* First crack at http RPC.

* Move client over to http RPC.

* Checkpoint. Remove proxy API.

* Use UNIX sockets.

* Remove obsolete file.

* Comments, minor improvements.

* Remove another file.

* LGTM tweaks.

richardkiss committed:
Fix flake8 warnings.

richardkiss committed:
Fix mypy warnings.

richardkiss committed:
Don't unlock lockfile until our port is open. Exit server cleanly.
Improve log message.

hoffmang9 committed:
Move netspace calculations to rpc-server. Refactor cli.

hoffmang9 committed:
Correct typing in get_total_miniters and get_network_space

hoffmang9 committed:
Update chiapos to 0.12.13

richardkiss committed:
Get rid of self._on_inbound_connect.

hoffmang9 committed:
Modify show for changes to /get_blockchain_status and place in CHANGELOG

hoffmang9 committed:
Adding Azure pipeline MacOS Mojave build runner (#222)

* Set up CI with Azure Pipelines
* Upgrade node w/ npm - break out deps, build, test
* Actually install the wheels for py.test to test
* Install timelord - requires boost - and upload wheels. Run all branches
* Skip p38 to punt on artifact conflict - add back dev reqs
* Add commentary

richardkiss committed:
Fix daemon problem: `start` now checks `is_running`.

richardkiss committed:
Remove unnecessary `serve_forever` task.

richardkiss committed:
Refactor a bunch of `self.` properties to tops of methods.

richardkiss committed:
Pass `global_connections` through the pipeline.

richardkiss committed:
Factor out `self._local_type` and `self._swrt_aiter`.

richardkiss committed:
Generate outbound_handshake once.

hoffmang9 committed:
Fix type errors in netspace rpc

richardkiss committed:
Add `log` property to `Connection` and use it.

richardkiss committed:
Removed every `self` from pipeline except in asserts and method invocation.

richardkiss committed:
Remove pipeline from `ChiaServer`.

richardkiss committed:
Factor out pipeline.

richardkiss committed:
Remove some instance variables from `ChiaServer` that it doesn't need.

hoffmang9 committed:
Bump PyYAML, cyrptography, aiosqlite to current versions

fchirica committed:
Fix compact pot protocol (#215)

* DB for challenge_hash. Refactor
* Store challenge_hash in headers db.
* Address review comments.

hoffmang9 committed:
chiapos to 0.12.14

hoffmang9 committed:
Use FetchContent to get pybind11 for chiavdf, chiapos, and blspy (#225)

Sub repos now require CMake 3.14+

hoffmang9 committed:
Updates to Timelord.md

richardkiss committed:
Remove a couple redundant items from requirements-dev.txt.

richardkiss committed:
If `CHIA_TEST_PLOT_ROOT` is set, use that path for plots in tests.

hoffmang9 committed:
bump chiapos, chiavdf, blspy for release

Yostra committed:
Electron react (#226)

* clean react
* add material ui
* add word list
* mnemonic v0
* jeepney backup
* keychain usage
* wallet api
* mnemonic ui
* mnemonics redux state
* handle exceptions correctly
* dashboard
* wallets
* get puzzle hash
* tx history
* sidebar
* start stop wallet node
* use existing mnemonics
* status info
* create cc wallet
* theme should be outside of switch
* create offer
* dbus alternative for linux
* key migration
* don't autocomplete, don't reset simulator db
* reset mnemonics
* Refactor keychain, and key migration
* Implement UI for changing keys
* Removing keys and mnemonic button
* Start making farmer and harvester RPCs
* start rpx for simulator
* Harvester and farmer websocket, and basic UI
* Plot display and deletion
* launch daemon on start
* State changes from full node, harvester, farmer, and full node ui improvements
* split balances in react
* pending change balance
* plotter
* dev config
* maintain connection / retry
* Remove electron-ui, and style fixes
* Better farmer and full node control
* Remove electron ui references
* Uncomment out starting the dameon
* Remove timelord tab, and change full node style
* Clean up new wallet login
* Refactor RPCs
* Now that the GH runner uses python 3.7.7 - use for windows installer
* add balance split to coloured coin wallet
* spendable balance fix
* Import private key from UI fix
* mac build/installer

Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>
Co-authored-by: Lipa Long <lipa@chia.net>
Co-authored-by: Gene Hoffman <hoffmang@hoffmang.com>

hoffmang9 committed:
Initial cleanups to CHANGELOG

mariano54 committed:
Ms.transaction fixes (#227)

* dbus alternative for linux

* key migration

* reset mnemonics

* Fix tests, and start working on key selection

* Implement UI for changing keys

* Removing keys and mnemonic button

* Start making farmer and harvester RPCs

* routing, daemon, react
* Fix rpc shutdown

* Harvester and farmer websocket, and basic UI
* Plot display and deletion
* use proper config:
* don't crash on nondefined
* local test false
* launch daemon on start
* farmer clean
* get blocks
* header table
* Start with farmer callback, fix bug in getting latest blocks
* State changes from full node, harvester, farmer, and full node ui improvements
* split balances in react
* pending change balance
* plotter
* dev config
* gitignore fail..
* maintain connection / retry
* Fix tests
* Remove electron-ui, and style fixes
* Better farmer and full node control
* Remove electron ui references
* Uncomment out starting the dameon
* Add to changelog
* Remove timelord tab, and change full node style
* Clean up new wallet login
* Refactor RPCs
* Fix flake8 import issues in full_node_rpc_server.py
* add balance split to coloured coin wallet
* spendable balance fix
* Import private key from UI fix
* Various fixes to transaction sending, and UI tweaks
* Fix merge issues

Co-authored-by: Yostra <straya@chia.net>
Co-authored-by: Lipa Long <lipa@chia.net>
Co-authored-by: Gene Hoffman <hoffmang@hoffmang.com>

hoffmang9 committed:
Clean up CHANGELOG

ronanlong committed:
split balance display

hoffmang9 committed:
Remove patch.txt, Detect python3.8 for timelord install

hoffmang9 committed:
Use python to find python version in timelord install

mariano54 committed:
Fix bad RPC for empty database (#228)

mariano54 committed:
Ms.fix balance (#229)

* Fix broken balance display and connection count display

mariano54 committed:
Fix plot logging display, second temp dir, and select field bug (#230)

Yostra committed:
don't use extended public key

Yostra committed:
Windows installer (#232)

* installer

* action

* remove this temporarely

* add logging

* python

* ico

* typo

* config file

* node installer

* remove unused var

* asar flag

* move setup events

* no asar

* closer to root

* move everything closer to home

* Revert "move everything closer to home"

This reverts commit 528083427aa59ba99b1fe9f302aa3b305f148cd5.

* Revert "closer to root"

This reverts commit f579fde829ceeb2b7ba5988dcbbc9228164428d9.

* long paths

* asa,r but unpack

* build mac on Azure

* azure

* don't use extended public key

mariano54 committed:
Fix create plots

Yostra committed:
Revert "Windows installer (#232)"

This reverts commit 1cb8840538294d5f65f7e0953c1a36bc4547f67b.

Yostra committed:
win insaller cleanup

Yostra committed:
delete wix

hoffmang9 committed:
install-timelord handles cmake, python3.8 - rename and clean build-ubuntu-mac.yml (#234)

* cleanup and rename build-ubuntu-mac.yml
* Remove vestiges of GH workaround for ubunutu/node and update Changelog
* timelord-install now handles cmake installation on Ubuntu b4 20.04LTS
* Bump chiavdf to fix timelord install, and bump chiapos for a fix

Yostra committed:
Breaking changes - fix aggsig, new genesis, new protocol version (#235)

* ouch - fix aggsig merge issues
* Block tools with keychain
* Use load_plots for Prover
* New genesis
* introducer
Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>

Yostra committed:
protocol bump

richardkiss committed:
New version of flake8 warns on f-strings without any "{}"
placeholders. This patch resolves the warnings.

Yostra committed:
use correct sk depending on pool pk

Yostra committed:
delete coinbase rewardS

Yostra committed:
typing

hoffmang9 committed:
Bump chiapos for Pi src distro issue, CHANGELOG on aggsig change

Yostra committed:
close on exit, don't open dev tolls if packaged

hoffmang9 committed:
Clean up installers (#239)

* Clean up win, mac installers
* Build Catalina DMG on GitHub Actions
* gitignore dmg builds
* Manage mac artifacts
* Correct Mac artifact, temp disable azure pytest
* Re-enable pytest on azure pipeline

Yostra committed:
min window size

hoffmang9 committed:
Place azure actions in venv - rename Mojave DMG, add npm fix to all (#241)

* Build in venv on azure
* Add npm audit fix, add venv to most steps on all runners
* Clean azure action, add npm fix to install.sh
* Rename azure pipeline Mojave DMG, prettify

hoffmang9 committed:
Remove duplicate npm installs, typos in ci's

Yostra committed:
ui touches

Yostra committed:
ping interval

Yostra committed:
undefinded store

Yostra committed:
lgtm

Yostra committed:
path change

Yostra committed:
update setup

Yostra committed:
start farmer and harvester after we have keys

Yostra committed:
after keys, longer pause

mariano54 committed:
Create plots filenames, chia keys API, and harvester reload plots fix (#243)

* Create plots filenames, chia keys API, and harvester reload plots fix
* Colon to hyphen
* Only refresh config file when reload_plots is called
* Use test root, and pass in root to wallet_node

Yostra committed:
hide farm button if not local test

Yostra committed:
fix simulator

Yostra committed:
prop name

Yostra committed:
Dont't consider eve coin as new transaction

mariano54 committed:
Fix farmer reward amount, correcly count fees

Yostra committed:
de dupe hash key pairs

Yostra committed:
black

hoffmang9 committed:
Adress some python lgtm items. Typo in changelog

hoffmang9 committed:
Fix import statements

Yostra committed:
gui version number

Yostra committed:
mac version

Yostra committed:
windows version

Yostra committed:
correct path

Yostra committed:
artifact names

hoffmang9 committed:
Capture BaseException.Exception instead of all of BaseException per lgtm

freddiecoleman commented on src/daemon/server.py:
```diff
@@ -0,0 +1,487 @@
+            self.log.info("Not implemented")
+
+        self.websocket_server = await websockets.serve(
+            self.safe_handle, "localhost", 55400
```
This will cause issues running from docker containers due to this issue https://github.com/Chia-Network/chia-blockchain/issues/219

Happy to workaround for now if you don't want to delay the release.

hoffmang9 replied:
I think you're going to still going to have to work around this in docker for 1.6 and we'll try to fix for 1.7.

hoffmang9 committed:
revert lgtm attempt

hoffmang9 committed:
revert websocket lgtm change - atom...

Yostra committed:
dmg name

Yostra committed:
Revert "de dupe hash key pairs"

This reverts commit e012f5b6187166e1299b92d37feb7f4345cfe41f.

ronanlong committed:
change cc wallet displayed balance units

hoffmang9 committed:
add wheel and pep517 to win pyinstall venv. Update npm pkgs

mariano54 committed:
Fix cc balance (#246)

mariano54 committed:
Display correct pending total balance

hoffmang9 committed:
Fix hidden aiohttp

hoffmang9 committed:
fix syntax error

hoffmang9 committed:
revert websocket_server spec entry.

hoffmang9 committed:
revert src.util.bip39

Yostra committed:
correct dir

mariano54 committed:
Fix double counting removals spendable balance (#247)

hoffmang9 committed:
Simplify build_windows with pip install --no-index --find-links (#248)

* only pip install chia-blockchain
* pip install from local index

matt-o-how committed:
changed text prompt for consistency
from "run chia keys generate_and_add" to "run chia keys generate" which 
is the actual command

matt-o-how committed:
added duplicate sigs where needed

matt-o-how committed:
Revert "added duplicate sigs where needed"

This reverts commit 5a4b7a022aeb22a5eb34bbd0fa4fc8167578119b.

Yostra committed:
use relative path:

hoffmang9 committed:
Add missing msvc dll's (#249)

* Bring in msvcp140.dll and vcruntime140_1.dll

hoffmang9 committed:
Put setproctitle and miniupnpc in the venv

hoffmang9 committed:
remove exteraneous --add-binary

hoffmang9 committed:
correct --find-links dir

hoffmang9 committed:
Clean up miniup-setproctitle

Yostra committed:
editbin

hoffmang9 committed:
Increase chiapos stack size with editbin.exe (#250)

* use mdbuild to path for editbin
* full path to editbin.exe
* because of course it's Start-Process...

Yostra committed:
tail force

Yostra committed:
correct name for check plots

Yostra committed:
typo

freddiecoleman commented on src/daemon/client.py:
```diff
@@ -0,0 +1,115 @@
+        if r["data"]["value"] == "pong":
+            return connection
+    except Exception as ex:
+        print(f"Exception {ex}")
```
During start up this logs the exception and looks like something is wrong but it is actually just a normal part of the current startup as it starts the daemon after it fails to connect the first time. It confused me for a bit as I tried to debug what was wrong.

Is it possible to allow the exception to throw and then it could be caught during start up and a slightly less worrying looking message could be logged.

hoffmang9 replied:
This one was annoying us too. Fixed here:
https://github.com/Chia-Network/chia-blockchain/pull/244/commits/f94f088166663160bf550c1baea1da25ab00d7af

ronanlong committed:
cc units in balance display

ronanlong committed:
fix spendable display double count

ronanlong committed:
remove below zero change to spendable balance

mariano54 committed:
Fix creating a plot and not having correct config on first launch

mariano54 committed:
Only print error if it is not connection refused

hoffmang9 committed:
Stop chia services on Windows #NoZombies (#253)

* simpler
* execsync
Co-authored-by: Strahinja Markovic <straya@chia.net>

mariano54 committed:
No dev tools open, bigger window, and warning for upnp

mariano54 committed:
Fix update after sync, and refresh after plot

hoffmang9 committed:
Correctly show pending but not spendable farming rewards (#256)

Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>

wjblanke committed:
add attribute error as fix for AttributeError: 'NoneType' object has no attribute 'resume_reading'

Yostra committed:
Code sign MacOS DMG, fix Mac icons, move to asar on mac install (#257)

* osx code signing in ci
* Install node
* asar and icon

hoffmang9 committed:
Update MacOS ci, fix wallet error message (#258)

* Build Mac on py 3.7 only, fix wallet error message

Yostra committed:
clean redux state on logout

Yostra committed:
delete cc wallet if not created succesfully

Yostra committed:
log rotate

hoffmang9 committed:
Update changelog for logrotate, asyncio bug

mariano54 committed:
Add tooltips for balance explanation and clean up code

mariano54 committed:
Fix crash when quickly switching walelts

matt-o-how committed:
Fix typo in pending state string

hoffmang9 committed:
Update changelog to Beta 1.6 release
